src/caffe/layers/batch_norm_layer.cpp

```diff
@@ -0,0 +1,230 @@
+#include <algorithm>
+#include <vector>
+
+#include "caffe/common_layers.hpp"
+#include "caffe/layer.hpp"
+#include "caffe/util/math_functions.hpp"
+
+namespace caffe {
+
+template <typename Dtype>
+void BatchNormLayer<Dtype>::LayerSetUp(const vector<Blob<Dtype>*>& bottom,
+      const vector<Blob<Dtype>*>& top) {
+  BatchNormParameter param = this->layer_param_.batch_norm_param();
```

Comment:
I don`t know why it does not work, I just quickly tested it on the same cifar_bn_sigmoid - just not converges at all :)